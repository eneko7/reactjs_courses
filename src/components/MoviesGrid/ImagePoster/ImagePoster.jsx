import React from 'react';
import PropTypes from 'prop-types';
import styles from './ImagePoster.scss';

const ImagePoster = ({
  src,
  alt,
}) => (
  <img src={src} alt={alt} className={styles.movieElementPoster} />
);

ImagePoster.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

export default ImagePoster;
