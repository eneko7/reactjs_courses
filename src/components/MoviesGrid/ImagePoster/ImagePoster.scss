@import "../../../assets/constants/constants.scss";

.movie_element_poster {
  width: 100%;
  height: 500px;
  object-fit: cover;
  margin-bottom: 20px;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
}

.poster {
  width: 35%;
  height: 525px;
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  margin-left: -3%;
}

@media (max-width: $laptop) {
  .movie_element_poster {
    height: 400px;
  }
}

@media (max-width: $tablet) {
  .movie_element_poster {
    height: 260px;
  }
}

@media (max-width: $phone) {
  .movie_element_poster {
    height: 350px;
  }
}
