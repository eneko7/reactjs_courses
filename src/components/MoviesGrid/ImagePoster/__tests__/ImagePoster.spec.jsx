import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ImagePoster from '../ImagePoster';

const renderer = new ShallowRenderer();
const data = {
  src: 'test',
  alt: 'test',
};
renderer.render(<ImagePoster {...data} />);
describe('ImagePoster', () => {
  it('renders correctly', () => {
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
