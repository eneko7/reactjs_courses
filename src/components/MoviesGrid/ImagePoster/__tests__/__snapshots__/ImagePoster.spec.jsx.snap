// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePoster renders correctly 1`] = `
<img
  alt="test"
  className="movie_element_poster"
  src="test"
/>
`;
